docs/streams/upgrade-guide.html

```diff
@@ -91,6 +91,29 @@ <h3><a id="streams_api_changes_280" href="#streams_api_changes_280">Streams API
         We extended <code>StreamJoined</code> to include the options <code>withLoggingEnabled()</code> and <code>withLoggingDisabled()</code> in
         <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-689%3A+Extend+%60StreamJoined%60+to+allow+more+store+configs">KIP-689</a>.
     </p>
+    <p>
+        We added two new methods to Kafka Streams, namely <code>addThread()</code> and <code>removeThread()</code> in
+        <a href="https://cwiki.apache.org/confluence/x/FDd4CQ">KIP-663</a>.
+        These enabled adding a removing StreamThreads to running KafkaStreams client.
+   </p>
+    <p>
+        We deprecated <code>setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler uncaughtExceptionHandler)</code>
+        in favor of <code>setUncaughtExceptionHandler(final StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler)</code>
+        in <a href="https://cwiki.apache.org/confluence/x/lkN4CQ">KIP-671</a>.
+        The default handler will close the client and the client will transit to state ERROR.
+        If you implement a custom handler, the new interface allows you to return a <code>StreamThreadExceptionResponse</code>,
+        which will determine how the application will respond to a thread failure.
+    </p>
+    <p>
+        Changes in <a href="https://cwiki.apache.org/confluence/x/FDd4CQ">KIP-663</a> necessitated the KafkaStreams client
+        state machine to update, which was done in <a href="https://cwiki.apache.org/confluence/x/lCvZCQ">KIP-696</a>..
+        The ERROR state is now terminal with PENDING_ERROR being a transitional state where the resources are closing.
+        The ERROR state indicates that there something wrong and should not be blindly restarted without classifying
```

Comment:
`that there something` -> `that there [is] something` ?

`and should not be blindly restarted` -> `and you should not restarted KafkaStreams blindly` ? (or `and [KafkaStreams] should not be blindly restarted`) ?

Reply:
I went with `is something wrong and the KafkaStreams clinet should not be blindly restarted`